### PR TITLE
refactor: migrating tables to use InteractiveTable component [DET-7382]

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description
Refactors the following components to use InteractiveTable instead of ResponsiveTable.

-   ModelDetails
-   ModelRegistry
-   TaskList
-   ExperimentTrials
-   JobQueue
-   TrialDetailsHyperparameters

ResponsiveTable is still in use in ClusterOverview because that component is no longer used in the new Clusters page, and HPTrialTable and TrialDetailsWorkloads due to difficulties caused by arbitrary columns.

## Test Plan
Manual testing by resizing and reordering columns in affected tables.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/pages/TaskList.tsx [hunk 1]
```diff
@@ -469,7 +498,7 @@ const TaskList: React.FC = () => {
           onAction={handleBatchAction}
           onClear={clearSelected}
         />
-        <ResponsiveTable<CommandTask>
```
Comment (hamidzr) [inline webui/react/src/pages/TaskList.tsx:472]: remove?

**Review by hamidzr — APPROVED** [review]
this looks good to me. thanks for updating these. I did some manual testing as well. can you add your test plan to the pr description?
Since this is changing so many of our views I'd suggest we land it after the release (if we have one this Wednesday) just because of the size of the change surface.

**Comment by hamidzr** [comment]
do we lose any of the responsive table features by going to this new comp?

File: webui/react/src/components/InteractiveTable.tsx [hunk 2]
```diff
@@ -32,7 +39,7 @@ type RecordType = any;
 export interface InteractiveTableSettings {
```
[the remark this answers is not in the document]
Comment (EmilyBonar) [reply webui/react/src/components/InteractiveTable.tsx:39]: 👍

File: webui/react/src/components/InteractiveTable.tsx [hunk 3]
```diff
@@ -52,6 +59,7 @@ interface ContextMenuProps {
 }
 
 export interface ColumnDef<RecordType> extends ColumnType<RecordType> {
```
[the remark this answers is not in the document]
Comment (EmilyBonar) [reply webui/react/src/components/InteractiveTable.tsx:70]: I think Trent might be addressing this in a future PR

File: webui/react/src/pages/JobQueue/JobQueue.table.tsx [hunk 4]
```diff
@@ -158,9 +174,10 @@ export const columns: ColumnType<Job>[] = [
   {
     align: 'right',
     className: 'fullCell',
+    dataIndex: 'action',
+    defaultWidth: DEFAULT_COLUMN_WIDTHS['action'],
```
[the remark this answers is not in the document]
Comment (EmilyBonar) [reply webui/react/src/pages/JobQueue/JobQueue.table.tsx:178]: InteractiveTable is currently set up in such a way that every column needs a `dataIndex` or it doesn't appear, even if that `dataIndex` doesn't refer to anything in the `dataSource`.

**Comment by EmilyBonar** [comment]
No, we shouldn't lose any features. InteractiveTable itself has already been in use on the experiment list for a while and performed well.

**Pull request merged**